chaco/tools/range_selection.py

```diff
@@ -591,7 +591,6 @@ def _set_selection(self, val):
 
             # Set the selection range on the datasource
             datasource.metadata[mdname] = val
-            datasource.metadata_changed = {mdname: val}
```

Comment:
it appears as though the `metadata` dict is being changed only once here, however it is being set potentially multiple times below.

Perhaps we only want the one change event to occur rather than being notified for all the changes.  This is the only place I am explicitly seeing where there may be a change in behavior from this.  If this does indeed change behavior, we will likely want to revert these removals.  

Perhaps this PR should simply be closed as `metadata_changed` seems here to stay.  I still do find it confusing and in any case we should more clearly document _why_ it exists, and how it is meant to be used differently than just listening to changes on the metadata directly (if that is in fact true)